### PR TITLE
Cleanup assorted Issues

### LINKED ISSUE #928: Potential Null Pointer Exception in 5 Places
1. Check out the latest code
2. Run FingBugs tool
3. Report potential null pointer exceptions
4. Analyze the code and confirm 5 places
---------------------------------------------
1. File name: GalleryGuiFactory.java
    Line: 189
    Changes: Should add `return;`, otherwise, a null comments object might trigger exception

2. File name: ObjectifyStorageIo.java
    Line: 427
    Line: 451
    Line: 474
    Changes: The code to update user data in memcache should be moved into `if (userData != null) { ... }`

3. File name: YoungAndroidProjectService.java
    Line: 691 & 692
    Changes: param buildURL could be null when buildErrorMsg(...) method is called. So, should put a check on buildURL param before using its .toString(...) method.

4. File name: GalleryList.java
    Line: 111
    Changes: Insert `apps = new ArrayList<GalleryApp>();`

5. File name: ProfilePage.java
    Line: 614
    Changes: Insert `keywordTotalResultsLabel = new Label(); container.add(keywordTotalResultsLabel);`

### PR DESCRIPTION
**Pull request by jisqyv**
Fix several minor issues (Fixes #928)

Change-Id: Idecd5a1eae1fadcf0b0db105ecc7c2ccc5afb499

### CODE REVIEW

File: appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProfilePage.java [hunk 1]
```diff
@@ -665,27 +665,18 @@ private void refreshApps(GalleryAppListResult appsResult, boolean refreshable) {
...
-    return apps.size();
-  }
-
-  /**
-   * Gets the number of selected projects
-   *
-   * @return the number of selected projects
+   * @return the number of selected apps
    */
   public int getNumSelectedApps() {
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProfilePage.java]: Ditto previous comment about unused methods.

File: appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryList.java [hunk 2]
```diff
@@ -468,27 +468,18 @@ private void refreshApps(GalleryAppListResult appsResult, int requestId, boolean
...
-    return apps.size();
-  }
-
-  /**
-   * Gets the number of selected projects
-   *
-   * @return the number of selected projects
+   * @return the number of selected apps
    */
   public int getNumSelectedApps() {
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryList.java]: According to IntelliJ, `getNumSelectedApps`, `getSelectedApps`, and `setSelectedTabIndex` are also not called on `master`, and it looks like they are not referenced on the branding branch either.

Comment (jisqyv) [reply appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryList.java]: @ewpatton I believe these functions are for future work... Let's leave them for now.

**Comment by jisqyv** [comment]
@ewpatton I refactored where the memcache.delete() is called.

File: appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryList.java (hunk 2 above)
Comment (ewpatton) [reply appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryList.java]: @jisqyv Sounds good to me.

**Pull request merged**